src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -112,47 +109,59 @@ class FilterButton extends React.Component<Props, State> {
         }
     };
 
-    handleFieldChange = (
-        index: number,
-        condition: Object,
-        fieldDisplayText: string | Date,
-        fieldDisplayTextType: string,
-        fieldKey: string | Date,
-        fieldKeyType: string,
-        valueType: string,
-    ) => {
+    handleColumnChange = (condition: Object, columnId: string) => {
+        const { columns } = this.props;
         const { conditions } = this.state;
-        const conditionToUpdate = conditions.find(currentCondition => {
+        let newConditionIndex = 0;
+        const conditionToUpdate = conditions.find((currentCondition, index) => {
+            newConditionIndex = index;
             return currentCondition.id === condition.id;
         });
-        if (conditionToUpdate) {
-            const updatedCondition = {
+
+        const column = columns && columns.find(c => c.id === columnId);
+
+        if (column) {
+            const type = column && column.type;
+
+            const operator = COLUMN_OPERATORS[type][0].key;
+
+            const newCondition = {
                 ...conditionToUpdate,
-                [fieldDisplayTextType]: fieldDisplayText,
-                [fieldKeyType]: fieldKey,
-                valueType,
+                columnId,
+                operator,
+                value: null,
             };
 
-            if (fieldKeyType === 'columnKey') {
-                // Upon selecting a new column, the operator and value fields should be reset.
-                updatedCondition.operatorKey = 0;
-                updatedCondition.operatorDisplayText = '';
-                updatedCondition.valueDisplayText = null;
-                updatedCondition.valueKey = null;
-            }
-
-            const updatedConditions = conditions.slice(0);
-            const conditionIndex = conditions.findIndex(
-                currentCondition => currentCondition.id === updatedCondition.id,
-            );
-            updatedConditions[conditionIndex] = updatedCondition;
+            const newConditions = conditions.slice(0);
+            newConditions[newConditionIndex] = newCondition;
 
             this.setState({
-                conditions: updatedConditions,
+                conditions: newConditions,
             });
         }
     };
 
+    handleFieldChange = (condition: Object, property: string | Date, conditionProperty: string) => {
```

Comment:
I'm not a fan of conditionProperty. I find this name confusing and line 154 undecipherable unless you are close to the code.